core/data/src/test/kotlin/com/google/samples/apps/nowinandroid/core/data/CompositeUserNewsResourceRepositoryTest.kt

```diff
@@ -42,70 +40,52 @@ class CompositeUserNewsResourceRepositoryTest {
 
     @Test
     fun whenNoFilters_allNewsResourcesAreReturned() = runTest {
-        // Obtain the user news resources flow.
+        // Obtain the user news resources flow without any filters applied.
         val userNewsResources = userNewsResourceRepository.observeAll()
 
-        // Send some news resources and user data into the data repositories.
+        // Send test data into the repositories.
         newsRepository.sendNewsResources(sampleNewsResources)
+        userDataRepository.setFollowedTopicIds(emptySet())
 
-        // Construct the test user data with bookmarks and followed topics.
-        val userData = emptyUserData.copy(
-            bookmarkedNewsResources = setOf(sampleNewsResources[0].id, sampleNewsResources[2].id),
-            followedTopics = setOf(sampleTopic1.id),
-        )
-
-        userDataRepository.setUserData(userData)
-
-        // Check that the correct news resources are returned with their bookmarked state.
+        // Check that the correct list of news resources is returned.
         assertEquals(
-            sampleNewsResources.mapToUserNewsResources(userData),
+            sampleNewsResources.mapToUserNewsResources(userDataRepository.userData.first()),
             userNewsResources.first(),
         )
     }
 
     @Test
     fun whenFilteredByTopicId_matchingNewsResourcesAreReturned() = runTest {
         // Obtain a stream of user news resources for the given topic id.
-        val userNewsResources =
-            userNewsResourceRepository.observeAll(
-                NewsResourceQuery(
-                    filterTopicIds = setOf(
-                        sampleTopic1.id,
-                    ),
-                ),
-            )
+        val userNewsResources = userNewsResourceRepository.observeAllForFollowedTopics()
```

Comment:
Changing the test in this way tests a different method: observeAllForFollowedTopics, which is tested in an identical way in the test below. 

This test should be left as-is. It tests the scenario where a user is not following any of those topics but is still viewing them. For example, when viewing the TopicDetail screen for a topic that they're not following.